### ISSUE TITLE
Datagrid doesn't resize when adding new items

### ISSUE THREAD

**Issue by redrabbit007** [AvaloniaUI/Avalonia#8771]
**Describe the bug**
I have a DataGrid inside of a StackPanel. The DataGrid.Items property is bound to a AvaloniaList in my view model. If I add new items to the list in my view model the height of the DataGrid grows when the first item is added, but it doesn't grow when more items are added. I can see the vertical scrollbar change its height with each item added. When I scroll down and back up then it resizes and shows all of the items like I would expect.

I also tried putting the DataGrid inside of a Grid with the row set to Auto with the same results.

**To Reproduce**

Create a new AvaloniaMVVM project.

In Nuget, update Avalonia packages to the latest (0.10.18).
In Nuget, Install the Avalonia.Controls.DataGrid package (0.10.18).

Add datagrid style include to App.axml.

```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="using:AvaloniaBugTest"
             x:Class="AvaloniaBugTest.App">
  <Application.DataTemplates>
    <local:ViewLocator/>
  </Application.DataTemplates>

  <Application.Styles>
    <FluentTheme Mode="Light"/>
    <StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Default.xaml"/>
  </Application.Styles>
</Application>
```

In MainWindow.axml add a StackPanel containing a Button and a DataGrid.

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaBugTest.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaBugTest.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaBugTest">

  <Design.DataContext>
    <vm:MainWindowViewModel/>
  </Design.DataContext>

  <StackPanel Orientation="Vertical">
    <Button
      Content="Add Item"
      Command="{Binding AddItemCommand}" />
    <DataGrid
      AutoGenerateColumns="True"
      Items="{Binding MyItems, Mode=OneWay}"
      VerticalScrollBarVisibility="Visible"
      GridLinesVisibility="All" />
  </StackPanel>

</Window>
```

In MainWindowViewModel:

```
using Avalonia.Collections;
using ReactiveUI;
using System;
using System.Reactive;

namespace AvaloniaBugTest.ViewModels
{
  public class MainWindowViewModel : ViewModelBase
  {
    public AvaloniaList<string> MyItems { get; } = new();

    public ReactiveCommand<Unit, Unit> AddItemCommand { get; }

    private readonly Random _random = new();

    public MainWindowViewModel()
    {
      AddItem();
      AddItem();
      AddItem();

      AddItemCommand = ReactiveCommand.Create(AddItem);
    }

    private void AddItem()
    {
      MyItems.Add("Some new item " + _random.Next());
    }
  }
}
```

Run the program. When you click the "Add Item" button the first time you will see the DataGrid grow in height, when you click on it more times it doesn't grow. If you scroll down and up it will show all of the items like expected.

**Expected behavior**
I expect the DataGrid to grow in height and show all of the items as they are added.

**Screenshots**
On startup
![image](https://user-images.githubusercontent.com/22695440/185249479-f2443ce3-05cb-4285-b74a-297fe1241585.png)

After adding 1 item.
![image](https://user-images.githubusercontent.com/22695440/185249497-4405d7ee-98a5-489f-ba9f-5fdbcc2e703a.png)

After adding lots of items.
![image](https://user-images.githubusercontent.com/22695440/185249527-2ced320f-d36d-4885-8b13-a2e8b0a63c10.png)

After scrolling
![image](https://user-images.githubusercontent.com/22695440/185249589-4b46d755-dd2e-4f76-b5f5-11893ffabc03.png)

**Desktop (please complete the following information):**
Windows
Avalonia Version 0.10.18.

**Additional context**
This seems related to https://github.com/AvaloniaUI/Avalonia/issues/2462 but I see the issue when using a StackPanel in addition to a Grid.

**Comment by timunie** [comment]
It's not reconmended to have a DataGrid with infinite height allowed, as it will break virtualization. But if your ItemsCount you expect is just very low, you can do this. In that case I'd disable vertical scrollviewer in DataGrid to make it auto grow.

**Comment by redrabbit007** [comment]
Ok, that makes sense, thank you.

Is there an easy way to disable the vertical scrollviewer in DataGrid?

**Comment by timunie** [comment]
use `ScrollViewer.[...]`- attached properties.

**Comment by cornerbowlsoftware** [comment]
@timunie I tried disabling as well as hiding the vertical scroll bar but after adding items the scroll bar shows up and is enabled.

ScrollViewer.VerticalScrollBarVisibility="Disabled"
and
ScrollViewer.VerticalScrollBarVisibility="Hidden"

```
<local:BaseMessagePreviewDataGrid
	Classes="dropdown"
    Grid.Column="0"
    x:Name="dataGrid"
    ItemsSource="{Binding ItemVMs}"
    HorizontalScrollBarVisibility="Auto"
	ScrollViewer.VerticalScrollBarVisibility="Disabled">
```

The datagrid is sitting inside an auto size grid row.  I tried putting it into a vertical stackpanel as well.  I am compiling with .Net 8

**Comment by timunie** [comment]
Autosize doesn't work for DataGrid. It needs a fixed size to work properly. Only option is a custom converter for height, based on ItemsCount

**Comment by grokys** [comment]
I think this is indeed a problem with `DataGrid` - even though virtualization is not expected to work when it's contained in a parent with no constraints, it should invalidate its measure when items are added or removed. Reopening.

**Comment by cornerbowlsoftware** [comment]
@maxkatz6 Is this fix in build 11.1.3?  Any pointers how to see the list of merges for the build that would be great.

**Comment by cornerbowlsoftware** [comment]
I see there is a new NuGet package 11.2.0-beta1.  I can confirm this now works as desired.  Thank you @grokys @emmauss 
 and @maxkatz6 .
